examples/ASViewController/Sample/DetailRootNode.m

```diff
@@ -0,0 +1,89 @@
+/* This file provided by Facebook is for non-commercial testing and evaluation
+ * purposes only.  Facebook reserves all rights not expressly granted.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * FACEBOOK BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#import "DetailRootNode.h"
+#import <AsyncDisplayKit/AsyncDisplayKit.h>
+
+#import "DetailCellNode.h"
+
+static const NSInteger kImageHeight = 200;
+
+@interface DetailRootNode () <ASCollectionViewDataSource, ASCollectionViewDelegate>
+@property (nonatomic, copy) NSString *imageCategory;
+@property (nonatomic, strong) ASCollectionNode *collectionNode;
+@end
+
+@implementation DetailRootNode
+
+#pragma mark - Lifecycle
+
+- (instancetype)initWithImageCategory:(NSString *)imageCategory
+{
+    self = [super init];
+    if (self == nil) { return self; }
+    
+    _imageCategory = imageCategory;
+    [self initNode];
+    
+    return self;
+}
+
+- (void)initNode
+{
+    // Create ASCollectionView. We don't have to add it explicitly as subnode as we will set usesImplicitHierarchyManagement to YES
+    UICollectionViewFlowLayout *layout = [[UICollectionViewFlowLayout alloc] init];
+    _collectionNode = [[ASCollectionNode alloc] initWithCollectionViewLayout:layout];
+    _collectionNode.delegate = self;
+    _collectionNode.dataSource = self;
+    _collectionNode.view.backgroundColor = [UIColor whiteColor];
+    
+    // Enable usesImplicitHierarchyManagement so the first time the layout pass of the node is happening all nodes that are referenced
+    // in layouts within layoutSpecThatFits: will be added automatically
+    self.usesImplicitHierarchyManagement = YES;
+}
+
+
+#pragma mark -
+
+- (ASLayoutSpec *)layoutSpecThatFits:(ASSizeRange)constrainedSize
+{
+    ASStaticLayoutSpec *staticSpec = [ASStaticLayoutSpec staticLayoutSpecWithChildren:@[self.collectionNode]];
```

Comment:
Since this is not used, here and in the other one you could just put it in the return; line.